tests/Feature/RequestPaginationTest.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+use GuzzleHttp\Promise\PromiseInterface;
+use Illuminate\Support\LazyCollection;
+use Saloon\Contracts\Response;
+use Saloon\Tests\Fixtures\Connectors\OffsetRequestPaginatorConnector;
+use Saloon\Tests\Fixtures\Connectors\PageRequestPaginatorConnector;
+use Saloon\Tests\Fixtures\Requests\OffsetGetSuperHeroesRequest;
+use Saloon\Tests\Fixtures\Requests\PageGetSuperHeroesRequest;
+
+test('you can configure a page paginator, and iterate over every request/response', function (): void {
+    $connector = new PageRequestPaginatorConnector;
+    $request = new PageGetSuperHeroesRequest;
+
+    $paginator = $connector->paginate($request);
+
+    $responses = [];
+    $superheroes = [];
+
+    foreach ($paginator as $response) {
+        $responses[] = $response;
+        $superheroes = [...$superheroes, ...$response->json('data')];
+    }
+
+    expect($responses)->toHaveCount(4)->each->toBeInstanceOf(Response::class)
+        ->and($superheroes)->toHaveCount(20)->each->toBeArray();
```

Comment:
Since the API is never going to change I would expect to actually see the right superheroes, because this may be doing the same page over and over again and your test wouldn't show it. Let me know if you want the JSON behind the API